src/c_api/c_api_ndarray.cc

```diff
@@ -278,6 +281,10 @@ int MXImperativeInvoke(AtomicSymbolCreator creator,
     }
   }
 
+  if (AutogradRuntime::Get()->IsRecording()) {
+    AutogradRuntime::Get()->RecordOp(op, attrs, ndinputs, ndoutputs);
```

Comment:
you need the Operator object created here https://github.com/dmlc/mxnet/pull/5129/files#diff-d63cbc91852e426d335d0dc65b25c9c6R231 for backward